internal/interceptor.go

```diff
@@ -213,6 +213,11 @@ type WorkflowOutboundInterceptor interface {
 	// AwaitWithTimeout intercepts workflow.AwaitWithTimeout.
 	AwaitWithTimeout(ctx Context, timeout time.Duration, condition func() bool) (bool, error)
 
+	// AwaitWithTimeoutWithOptions intercepts workflow.AwaitWithTimeoutWithOptions.
+	//
+	// NOTE: Experimental
+	AwaitWithTimeoutAndOptions(ctx Context, timeout time.Duration, options TimerOptions, condition func() bool) (bool, error)
```

Comment:
I wonder if we should just call this `AwaitWithOptions` and take timeout as an optional option?

Reply:
As in move `timeout` into the `TimerOptions` struct? If that's what you mean, do we want to change the `NewTimerWithOptions` to match?

Reply:
More like create `AwaitOptions` and put `timeout` and `TimerOptions` under it. This is so if we need to add more fields we don't have to add more overrides